lib/core.coffee

```diff
@@ -1,102 +1,66 @@
-      return true
-    if major is 1 and minor > 9
-      return true
-    if major is 1 and minor is 9 and patch >= 5
-      return true
-    false
+configImportsModules = (config) ->
+  return true for ruleName, rconfig of config when rconfig.module?
+  return userConfig?.coffeelint?.transforms?
 
-  isCompatibleWithAtom: (coffeelint) ->
-    [major, minor, patch] = coffeelint.VERSION.split('.').map(toInt)
+module.exports = (filePath, source, isLiterate) ->
+  showUpgradeError = false
 
-    if major > 1
-      return true
-    if major is 1 and minor > 9
-      return true
-    if major is 1 and minor is 9 and patch >= 1
-      return true
-    false
+  coffeeLintPath = _resolveCoffeeLint(filePath)
+  coffeelint = require(coffeeLintPath)
 
-  lint: (filePath, source, scopeName) ->
-    isLiterate = scopeName is 'source.litcoffee'
-    showUpgradeError = false
-
-    coffeeLintPath = @_resolveCoffeeLint(filePath)
+  # Versions before 1.9.1 don't work with Atom because of an assumption that
+  # if window is defined, then it must be running in a browser. Atom breaks
+  # this assumption, so CoffeeLint < 1.9.1 will fail to find CoffeeScript.
+  # See https://github.com/clutchski/coffeelint/pull/383
+  if semver.lt(coffeelint.VERSION, '1.9.1')
+    coffeeLintPath = 'coffeelint'
     coffeelint = require(coffeeLintPath)
+    showUpgradeError = true
 
-    # Versions before 1.9.1 don't work with atom because of an assumption that
-    # if window is defined, then it must be running in a browser. Atom breaks
-    # this assumption, so CoffeeLint < 1.9.1 will fail to find CoffeeScript.
-    # See https://github.com/clutchski/coffeelint/pull/383
-    [major, minor, patch] = coffeelint.VERSION.split('.').map(toInt)
-    if not @isCompatibleWithAtom(coffeelint)
-      coffeeLintPath = 'coffeelint'
-      coffeelint = require(coffeeLintPath)
-      showUpgradeError = true
-
-    configFinder = require("#{coffeeLintPath}/lib/configfinder")
+  configFinder = require("#{coffeeLintPath}/lib/configfinder")
 
-    result = []
-    try
-      config = configFinder.getConfig(filePath)
-      if @configImportsModules(config) and not @canImportModules(coffeelint)
-        showUpgradeError = true
-      else
-        result = coffeelint.lint(source, config, isLiterate)
-    catch e
-      console.log(e.message)
-      console.log(e.stack)
-      result.push({
-        lineNumber: 1
-        level: 'error'
-        message: "CoffeeLint crashed, see console for error details."
-        rule: 'none'
-      })
+  result = []
+  try
+    config = configFinder.getConfig(filePath)
+    if configImportsModules(config) and semver.lt(coffeelint.VERSION, '1.9.5')
+      showUpgradeError = true
+    else
+      result = coffeelint.lint(source, config, isLiterate)
+  catch e
+    console.log(e.message)
+    console.log(e.stack)
+    result.push({
+      lineNumber: 1
+      level: 'error'
+      message: 'CoffeeLint crashed, see console for error details.'
+      rule: 'none'
```

Comment:
I'm wondering if we should just let this throw a error instead of showing this on line 1